cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -780,11 +781,17 @@ private String asLabel(String val) {
 
   /**
    * Kubernetes names must be lowercase alphanumeric, or '-'. Some are less restrictive, but those
-   * characters should always be ok.
+   * characters should always be ok. They should also start and end with an alphanumeric.
    */
-  private String cleanse(String val, int maxLength) {
+  @VisibleForTesting
+  static String cleanse(String val, int maxLength) {
+    // Replace characters that are not alphanumeric or '=' with '-'
     String cleansed = val.replaceAll("[^A-Za-z0-9\\-]", "-").toLowerCase();
-    return cleansed.length() > maxLength ? cleansed.substring(0, maxLength) : cleansed;
+    // Truncate to maxLength if needed
+    cleansed = cleansed.length() > maxLength ? cleansed.substring(0, maxLength) : cleansed;
+    // Remove leading and trailing '-'
+    cleansed = StringUtils.stripEnd(cleansed, "-");
+    return StringUtils.stripStart(cleansed, "-");
```

Comment:
some systems have more restrictions on the start character (like it can't be a number). Can you double check that we're not missing anything else for the first char?

Reply:
Verified that pipeline with only numbers in the name was able to run succesfully